spec/lib/funding_counter_spec.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'FundingCounter' do
+  let(:page) { create(:page, total_donations: 10_000.00) }
+
+  describe 'money_conversions' do
+    let(:record) { FundingCounter.new(page, 'USD', 20) }
+
+    it 'should calculate original_amount' do
+      expect(record.original_amount.to_f).to eql 100.0
+    end
+
+    it 'should calculate converted_amount' do
+      expect(record.converted_amount.to_f).to eql 20.0
+    end
+  end
+
+  describe '.update' do
+    before do
+      FundingCounter.new(page, 'USD', 20).update
+    end
+
+    it 'should increment page donations by 20' do
+      page.reload
+      expect(page.total_donations.to_f).to eql 12_000.0
+    end
+  end
+
+  describe '.update' do
+    before do
+      FundingCounter.new(page, 'USD', 10, true).update
+    end
```

Comment:
not sure I'm convinced by this API. I would expect something along the lines of `FundingCounter.new(page).update(-10, 'USD')` or, in keeping with the current approach, `FundingCounter.new(page, 'USD', -10)` if we're decrementing.

I do believe that calls should be expressive and intuitive.